common.py

```diff
@@ -400,54 +399,77 @@ def prepare_filelist(self, names, path, goto, indent):
         items += files
         return items
 
-    def is_hidden(self, filename, path, goto=''):
+    def is_hidden(self, item, path, goto='', get_is_dir=False, dirs_only=False):
         if not (path or goto):  # special case for ThisPC
             return False
-        tests = self.view.settings().get('dired_hidden_files_patterns', ['.*'])
-        if isinstance(tests, str):
-            tests = [tests]
-        if any(fnmatch.fnmatch(filename, pattern) for pattern in tests):
-            return True
-        if sublime.platform() != 'windows':
-            return False
-        # check for attribute on windows:
-        try:
-            attrs = ctypes.windll.kernel32.GetFileAttributesW(join(path, goto, filename))
-            assert attrs != -1
-            result = bool(attrs & 2)
-        except (AttributeError, AssertionError):
-            result = False
-        return result
-
-    def try_listing_directory(self, path):
+        show_hidden = self.show_hidden
+        show_excluded = self.view.settings().get('dired_show_excluded_files', True)
+        is_hidden = False
+        filename = item['file']
+        fullpath = join(path, goto, filename)
+        is_dir = item['is_dir']
+        result = lambda: is_hidden if not get_is_dir else [is_hidden, is_dir]
+        if dirs_only and not is_dir:
+            return result()
+        if not is_hidden and not show_excluded:
+            if is_dir:
+                tests = self.view.settings().get('folder_exclude_patterns', [])
+                if any(fnmatch.fnmatch(filename, p) for p in tests):
+                    is_hidden = True
+            else:
+                tests = self.view.settings().get('file_exclude_patterns', [])
+                if any(fnmatch.fnmatch(filename, p) for p in tests):
+                    is_hidden = True
+        if not is_hidden and not show_hidden:
```

Comment:
Regarding this you said earlier

    > In case `self.show_hidden` is `True` the method `is_hidden` must not be called at all, we check it before calling `os.listdir()` in other methods, e.g. `traverse_tree`

But I see that `is_hidden` is tested (only?) in [`try_listing_directory`](../../blob/master/common.py#L430)


[`traverse_tree`](../../blob/master/dired.py#L240-278) has neither checks, you meant `try_listing_directory` only, right?

I thought it would be better to have the is_hidden/is_excluded check all at one place. Thoughts?